File: plugin/src/test/resources/golden/go/address.go
package example

import (
	"encoding/json"
	"fmt"
)

type Address struct {
	Street string
	City string
	State string
	Zip int32
}

func (m *Address) Serialize() []any {
	arr := make([]any, 4)
	arr[0] = m.Street
	arr[1] = m.City
	arr[2] = m.State
	arr[3] = m.Zip
	return arr
}

func (m *Address) Marshal() ([]byte, error) {
	arr := m.Serialize()
	data, err := json.Marshal(arr)
	if err != nil {
		return nil, err
	}
	return data, nil
}

func DeserializeAddress(arr []any) (*Address, error) {
	obj := &Address{}
	size := len(arr)
	if size > 0 && arr[0] != nil {
		if v, ok := arr[0].(string); ok {
			obj.Street = v
		}
	}
	if size > 1 && arr[1] != nil {
		if v, ok := arr[1].(string); ok {
			obj.City = v
		}
	}
	if size > 2 && arr[2] != nil {
		if v, ok := arr[2].(string); ok {
			obj.State = v
		}
	}
	if size > 3 && arr[3] != nil {
		if v, ok := arr[3].(float64); ok {
			obj.Zip = int32(v)
		}
	}
	return obj, nil
}

func (m *Address) Unmarshal(data []byte) error {
	var arr []any
	if err := json.Unmarshal(data, &arr); err != nil {
		return fmt.Errorf("failed to unmarshal JSON: %w", err)
	}
	parsed, err := DeserializeAddress(arr)
	if err != nil {
		return err
	}
	*m = *parsed
	return nil
}

File: plugin/src/test/resources/golden/swift/Address.swift
import Foundation

public struct Address {
    public var street: String = ""
    public var city: String = ""
    public var state: String = ""
    public var zip: Int32 = 0

    public init() {
    }

    public func serialize() -> [Any] {
        var arr: [Any] = Array(repeating: NSNull(), count: 4)
        arr[0] = street
        arr[1] = city
        arr[2] = state
        arr[3] = zip
        return arr
    }

    public func serializedData() throws -> Data {
        let arr = serialize()
        return try JSONSerialization.data(withJSONObject: arr, options: [])
    }

    public static func deserialize(_ arr: [Any?]) -> Address {
        var obj = Address()
        let size = arr.count
        if size > 0, let elem = arr[0] {
            if let v = elem as? String {
                obj.street = v
            }
        }
        if size > 1, let elem = arr[1] {
            if let v = elem as? String {
                obj.city = v
            }
        }
        if size > 2, let elem = arr[2] {
            if let v = elem as? String {
                obj.state = v
            }
        }
        if size > 3, let elem = arr[3] {
            if let v = elem as? Double {
                obj.zip = Int32(v)
            }
        }
        return obj
    }

    public init(serializedData data: Data) throws {
        guard let arr = try JSONSerialization.jsonObject(with: data, options: []) as? [Any] else {
            throw NSError(domain: "Address", code: -1, userInfo: [NSLocalizedDescriptionKey: "Expected JSON array"])
        }
        let parsed = Address.deserialize(arr.map { $0 is NSNull ? nil : $0 })
        self = parsed
    }
}
